sql/src/main/java/io/crate/executor/transport/SingleRowSingleValueRowReceiver.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.executor.transport;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.SettableFuture;
+import io.crate.analyze.symbol.SelectSymbol;
+import io.crate.concurrent.CompletionListenable;
+import io.crate.core.collections.Row;
+import io.crate.operation.projectors.*;
+import io.crate.planner.Plan;
+
+import java.util.Set;
+
+/**
+ * RowReceiver expects to receive only one row and triggers a future with the value once completed
+ */
+class SingleRowSingleValueRowReceiver implements RowReceiver, CompletionListenable {
```

Comment:
probably, but I'd like to keep it package private until some other package needs it